### PR TITLE
aria-label for external links

### LINKED ISSUE #1873: Update external inline links to use an aria-label
**Is your feature request related to a problem? Please describe.**
A decision was made to apply the **', external'** label via visually hidden text, this means that this text is now part of the article text when CSS hasn't loaded/turned off, not only does this look odd, this part of the article text may no longer make complete sense. Please use an aria-label instead.

Current implementation:
`<a href="my-link.co.uk">Israel <span class="visually-hidden">, external</span></a>`

**Describe the solution you'd like**
`<a href="my-link.co.uk" aria-label="Israel, external">Israel</a>`

**Describe alternatives you've considered**

**Testing notes**
Testing both the standard link which is external has the correct markup (http://localhost:7080/news/articles/c0000000006o) and regression testing the react-router link  (a link that takes you to another Simorgh Article page) does not change. e.g. first link on here: http://localhost:7080/news/articles/cn7769kpk9mo

Dev insight: Unit tests will be needed. 

**Additional context**

### PR DESCRIPTION
**Pull request by twinlensreflex**
Resolves #1873

**Overall change:** 
Adds logic so that an `aria-label` is used to show that a link is external

**Code changes:**
- Aria-label added to `InlineLink` container
- Snapshot tests updated

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/InlineLink/index.jsx [hunk 1]
```diff
@@ -32,13 +32,23 @@ const InlineLinkContainer = ({ locator, isExternal, blocks }) => {
     );
   }
 
-  // else return a normal hyperlink
+  if (isExternal) {
+    return (
+      <InlineLink
+        href={locator}
+        aria-label={`${blocks[0].model.text}${externalLinkText}`}
```
Comment (twinlensreflex) [inline src/app/containers/InlineLink/index.jsx]:
```suggestion
        aria-label={`${blocks[0].model.text}${externalLinkText}`}
```

this line isn't really ideal but i'm not sure what I should be doing here

Comment (12) [reply src/app/containers/InlineLink/index.jsx]: At the very least I'd suggest constructing it before the return and then just referencing it here.

File: src/app/containers/InlineLink/index.jsx [hunk 2]
```diff
@@ -32,13 +32,23 @@ const InlineLinkContainer = ({ locator, isExternal, blocks }) => {
     );
   }
 
-  // else return a normal hyperlink
+  if (isExternal) {
```
Comment (12) [inline src/app/containers/InlineLink/index.jsx]: I don't think you need this condition. You could have the single return block as before, and just do

```js
<InlineLink href={locator} aria-label={isExternal ? `${blocks[0].model.text}${externalLinkText}`: null}>
```

Comment (sareh) [reply src/app/containers/InlineLink/index.jsx]: To make this clearer to read, defining the variable above would be better & using deepGet for the text:
```
const linkText = deepGet([0, 'model', 'text'], blocks);
const ariaLabel= isExternal ? `${linkText}${externalLinkText}`: null;
```

File: src/app/containers/InlineLink/index.jsx [hunk 3]
```diff
@@ -32,13 +32,23 @@ const InlineLinkContainer = ({ locator, isExternal, blocks }) => {
...
+    return (
+      <InlineLink
+        href={locator}
+        aria-label={`${blocks[0].model.text}${externalLinkText}`}
+      >
+        <Blocks blocks={blocks} componentsToRender={componentsToRender} />
+        {isExternal ? (
+          <VisuallyHiddenText>{externalLinkText}</VisuallyHiddenText>
```
Comment (sareh) [inline src/app/containers/InlineLink/index.jsx]: Also, the VisuallyHiddenText component can be removed from here, since it is no longer required.

**Review by RayNjeri — APPROVED** [review]
LGTM

**Review by oluoluoxenfree — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM.

**Pull request merged**